marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java

```diff
@@ -657,9 +622,11 @@ private void runInDatabase(String query, String databaseName) {
         commandMap.put("mlSecurityCommands", securityCommands);
 
         List<Command> dbCommands = new ArrayList<>();
-        dbCommands.add(new DeployHubOtherDatabasesCommand(hubConfig));
-        dbCommands.add(new DeployHubFinalTriggersDatabaseCommand(hubConfig));
-        dbCommands.add(new DeployHubFinalSchemasDatabaseCommand(hubConfig));
+        // TODO Hoping we don't need anything here, because the staging command list has everything we need
+        // Ideally we can get rid of the concept of "staging" commands and "final" commands
+//        dbCommands.add(new DeployHubOtherDatabasesCommand(hubConfig));
+//        dbCommands.add(new DeployHubFinalTriggersDatabaseCommand(hubConfig));
+//        dbCommands.add(new DeployHubFinalSchemasDatabaseCommand(hubConfig));
```

Comment:
The database files in the path "ml-config/databases/" (final, final-schemas, final-triggers, modules) are not getting deployed. I think we may have to include this snippet to deploy those files

DeployOtherDatabasesCommand databaseCommand = new DeployOtherDatabasesCommand();
        dbCommands.add(new DeployHubStagingTriggersDatabaseCommand(hubConfig));	        databaseCommand.setDeployDatabaseCommandFactory(new HubDeployCommandDatabaseFactory(hubConfig));
        dbCommands.add(new DeployHubStagingSchemasDatabaseCommand(hubConfig));	        dbCommands.add(databaseCommand);

Reply:
@srinathgit Have you tried a build of this? I did "gradle publishtomavenlocal" with version = 4.0.3.1 and then used it on a new DHF project; the final databases are all deployed correctly.

Reply:
Installation using java api seems to have failed in jenkins with the following message https://jenkins.marklogic.com/job/dhf-core-develop-rh7-PR/1759/consoleFull

11:32:36.275 [main] INFO  c.m.h.d.c.DeployUserServersCommand - Processing file: /space/jenkins/workspace/dhf-core-develop-rh7-PR/marklogic-data-hub/ye-olde-project/src/main/ml-config/servers/final-server.json
Exception in thread "main" 11:32:36.669 [main] ERROR c.m.r.util.MgmtResponseErrorHandler - Logging HTTP response body to assist with debugging: {"errorResponse":{"statusCode":"400", "status":"Bad Request", "messageCode":"MANAGE-INVALIDPAYLOAD", "message":"MANAGE-INVALIDPAYLOAD: (err:FOER0000) Payload has errors in structure, content-type or values. No such database"}}
11:32:36.670 [main] ERROR c.m.m.r.appservers.ServerManager - Error occurred while sending POST request to /manage/v2/servers; logging request body to assist with debugging: {"server-name":"data-hub-FINAL","server-type":"http","root":"/","group-name":"Default","port":8011,"modules-database":"data-hub-MODULES","content-database":"data-hub-FINAL","authentication":"digest","default-error-format":"json","error-handler":"/MarkLogic/rest-api/error-handler.xqy","url-rewriter":"/MarkLogic/rest-api/rewriter.xml","rewrite-resolves-globally":true}

Using your branch, I added that snippet in getFinalCommands() and installation/ uninstallation ran fine in my local

Reply:
I believe this issue is because of the separation of getStagingCommands and getFinalCommands, with the latter being run first and then followed by the former. So with two separate lists, getStagingCommands doesn't need a set of database commands but getFinalCommands does (if both have the database commands, then the database files will all get processed twice).

On my other branch, I'm experimenting with just one set of commands and I'm not having any problems so far.